src/data_source/sql.rs

```diff
@@ -586,20 +622,25 @@ impl Drop for SqlDataSource {
 }
 
 #[async_trait]
-impl VersionedDataSource for SqlDataSource {
+impl<Types, I> VersionedDataSource for SqlDataSource<Types, I>
+where
+    Types: NodeType,
+    I: NodeImplementation<Types>,
+    Block<Types>: QueryableBlock,
+{
     type Error = postgres::error::Error;
 
     /// Atomically commit to all outstanding modifications to the data.
     ///
     /// If this method fails, outstanding changes are left unmodified. The caller may opt to retry
     /// or to erase outstanding changes with [`revert`](Self::revert).
     async fn commit(&mut self) -> Result<(), Self::Error> {
-        if !self.tx_in_progress {
-            // Nothing to do if there is no outstanding transaction.
-            return Ok(());
+        if self.tx_in_progress {
+            self.client.batch_execute("COMMIT").await?;
```

Comment:
There are currently no tests failing for me if I remove the two lines 639-40.

Reply:
How did you even notice this!

This would be because all the tests use the same SQL connection for all queries, so the queries are all executing against the same uncommitted transaction. I will add a test that the results of some updates are visible even if creating an entirely new database connection

Reply:
Added a check in [b919351](https://github.com/EspressoSystems/hotshot-query-service/pull/249/commits/b9193510bab679fe9ed3fa3cd98328855759ba04) that fails if these lines are commented out